modules/dnn/src/net_impl.cpp

```diff
@@ -1410,8 +1410,8 @@ void Net::Impl::setInput(InputArray blob, const String& name, double scalefactor
             const MatShape& inputShapeLimitation = netInputLayer.shapes[pin.oid];
             if (!inputShapeLimitation.empty())
             {
-                CV_CheckEQ(inputShapeLimitation.size(), blobShape.size(), "");
 #if 0  // TODO: DNNTestNetwork.MobileNet_SSD_Caffe_Different_Width_Height/0
+                CV_CheckEQ(inputShapeLimitation.size(), blobShape.size(), "");
```

Comment:
It's not relevant change.

Reply:
The following tests were failed:
```cpp
[  FAILED  ] Test_ONNX_layers.Convolution_variable_weight_bias/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.Conv1d_variable_weight_bias/0, where GetParam() = OCV/CPU
```
```
C++ exception with description "OpenCV(4.7.0-dev) /home/dkurt/opencv/modules/dnn/src/net_impl.cpp:1413: error: (-2:Unspecified error) in function 'virtual void cv::dnn::dnn4_v20221220::Net::Impl::setInput(cv::InputArray, const String&, double, const Scalar&)'
>  (expected: 'inputShapeLimitation.size() == blobShape.size()'), where
>     'inputShapeLimitation.size()' is 1
> must be equal to
>     'blobShape.size()' is 2
" thrown in the test body.
```
The thing is that there is no limitation for the shapes - OpenCV networks are flexible for any valid input size,

Reply:
It makes sense to hide/remove `inputShapeLimitation` constant with the condition. It's abandoned constant with empty condition.

Reply:
Decided to hide it under `#if 0` completely.